src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Download/PackageDownloadRunner.cs

```diff
@@ -0,0 +1,316 @@
+            {
+                logger.LogMinimal(string.Format(
+                    CultureInfo.CurrentCulture,
+                    Strings.PackageDownloadCommand_Starting,
+                    package.Id,
+                    string.IsNullOrEmpty(package.NuGetVersion?.ToNormalizedString()) ? Strings.PackageDownloadCommand_LatestVersion : package.NuGetVersion.ToNormalizedString()));
+
+                try
+                {
+                    (NuGetVersion version, SourceRepository downloadRepository) =
+                        await ResolvePackageDownloadVersion(
+                            package,
+                            sourceRepositories,
+                            cache,
+                            logger,
+                            args.IncludePrerelease,
+                            token);
+
+                    if (version == null)
+                    {
+                        // Unable to find a valid version
+                        downloadedAllSuccessfully &= false;
+                        continue;
+                    }
+
+                    bool download = await DownloadPackageAsync(
+                                        package.Id,
+                                        version,
+                                        downloadRepository,
+                                        cache,
+                                        settings,
+                                        outputDirectory,
+                                        logger,
+                                        token);
+
+                    if (download)
+                    {
+                        logger.LogMinimal(string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.PackageDownloadCommand_Succeeded,
+                            package.Id,
+                            version,
+                            outputDirectory));
+                    }
+                    else
+                    {
+                        logger.LogError(string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.PackageDownloadCommand_Failed,
+                            package.Id,
+                            version));
+
+                        downloadedAllSuccessfully &= false;
+                    }
+                }
+#pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception ex)
+                {
+                    logger.LogError(ex.ToString());
+                    downloadedAllSuccessfully &= false;
+                }
+#pragma warning restore CA1031 // Do not catch general exception types
+            }
+
+            return downloadedAllSuccessfully ? ExitCodeSuccess : ExitCodeError;
+        }
+
+        internal static async Task<(NuGetVersion, SourceRepository)> ResolvePackageDownloadVersion(
+            PackageWithNuGetVersion package,
```

Comment:
```suggestion
            PackageWithNuGetVersion packageToResolve,
```

Renaming this will let you avoid having to use `p` as a variable name later on